DAOS-2657 control: refactor result maps using fmt.Stringer

tanabarr opened the pull request:
Instead of using type assertion to apply behaviour to a generic
parameter, implement fmt.Stringer interface on specific types
enabling representation through polymorphic function String().

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

daosbuild1 commented:
Test stage Build RPM on SLES 12.3 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-610/1/execution/node/235/log

daosbuild1 commented:
Test stage Build RPM on SLES 12.3 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-610/2/execution/node/203/log

tanabarr commented:
I don't really want to increase our version dependency to 1.10 from 1.9 just to use strings.Builder. I refactored to use bytes.Buffer instead and fixed issues where go unit tests were intermittently failing due to the key/values in maps being evaluated in an arbitrary order and therefore not matching the expected fixtures.
